Common/src/Pollster/TaskHandler.cs

```diff
@@ -594,10 +594,36 @@ await agent_.FinalizeTaskCreation(CancellationToken.None)
                                              TaskData          taskData,
                                              CancellationToken cancellationToken)
   {
-    if (cancellationToken.IsCancellationRequested || e is RpcException
-                                                          {
-                                                            StatusCode: StatusCode.Unavailable,
-                                                          })
+    await HandleErrorInternalAsync(e,
+                                   taskData,
+                                   false,
+                                   true,
+                                   cancellationToken)
+      .ConfigureAwait(false);
+  }
+
+  private async Task HandleErrorResubmitAsync(Exception         e,
+                                              TaskData          taskData,
+                                              CancellationToken cancellationToken)
+  {
+    await HandleErrorInternalAsync(e,
+                                   taskData,
+                                   true,
+                                   false,
+                                   cancellationToken)
+      .ConfigureAwait(false);
+  }
+
+  private async Task HandleErrorInternalAsync(Exception         e,
+                                              TaskData          taskData,
+                                              bool              resubmit,
+                                              bool              requeueIfUnavailable,
```

Comment:
`requeueIfUnavalible` does not seem to be used. However, the refactor changed the status of the queue message.

Reply:
It is used.
Good point ! I'll fix it.